pkgs/top-level/all-packages.nix

```diff
@@ -14793,11 +14793,11 @@ in
 
   inherit (callPackages ../development/libraries/libressl { })
     libressl_3_0
```

Comment:
LibreSSL 3.0 is now no longer supported, can you please remove it?

LibreSSL releases are supported for one year after the corresponding OpenBSD release, which was [6.6](https://www.openbsd.org/66.html) for LibreSSL 3.0, and OpenBSD 6.6 is now more than one year old.

Reply:
Thank you for reviewing! Currently `wasm-pack` depends on it:
```nix
  wasm-pack = callPackage ../development/tools/wasm-pack {
    # Wasm-pack depends on a version of rust-openssl which is incompatible with
    # LibreSSL 3.1, so we explicitly opt for the older version.
    libressl = libressl_3_0;
    inherit (darwin.apple_sdk.frameworks) Security;
  };
```
so we need investigations but I agree with your point.

Reply:
Oh, right, I missed that one in my `git grep`.

The wasm-pack issue is caused by rust-openssl explicitly rejecting unknown LibreSSL versions (see also https://github.com/sfackler/rust-openssl/issues/1278). [It looks like they don’t even support 3.2.1 yet](https://github.com/sfackler/rust-openssl/pull/1333), so it will probably be a while before rust-openssl, and by extension wasm-pack, can build with LibreSSL 3.2. But maybe it gained support for 3.1 since that line was added?

If not, maybe we can patch rust-openssl to remove the check. If I understand correctly, the reason that the author added an explicit check to reject unknown versions is because LibreSSL does not expose a stable ABI, so a version of rust-openssl built for 3.1 might break when used with 3.2. But we don’t have this problem with Nix, because using 3.2 will cause a rebuild of rust-openssl.

Reply:
Agreed.

Reply:
Now [rust-openssl supports LibreSSL up to 3.3.1](https://github.com/sfackler/rust-openssl/blob/openssl-v0.10.32/openssl-sys/build/main.rs#L224-L228). I will revisit again soon(-ish).

Reply:
Addressed on https://github.com/NixOS/nixpkgs/pull/114180. Thank you!